Add navigation button to project preview image

psalmuelle opened the pull request:
## Summary
Users can now have access to the next and previous buttons while previewing project images. There is also an explicit cancel button on the top right corner of the images.
If project image is one, only the image is shown as there is no need for a carousel.

Closes #435 

## Changes

- Replaced preview image with slider in the preview dialog
- Changed image dialog style
- Added the cancel button to the dialog

## Screenshots

![Screenshot 2023-03-09 103357](https://user-images.githubusercontent.com/81099447/223980760-e262f384-ccaa-423d-bdd5-8fc1dc5f25c1.jpg)

psalmuelle committed:
feat: make project preview image a carousel

psalmuelle committed:
Merge branch 'master' of https://github.com/psalmuelle/zubhub

tuxology commented:
Thanks @psalmuelle 🤝  Some suggestions:
 - Can we place the red cross on the image corner itself, something like this? 
![image](https://user-images.githubusercontent.com/3373857/224174391-8235bcd0-2418-4add-b510-46832b3792d6.png)
 - In addition, we should also improve the left/right arrows styling a bit more so that they are clearly visible. I think this can also be achieved by decreasing opacity of the modal background (make it more darker - maybe 70% dark)
 - Another suggestion is to increase the size of the image so it takes up more screen size. The issue we see currently is also that image size shown even after clicking is very small. 

Finally, please test this on responsive views as well and share screenshots. After this, we can star code review.

psalmuelle commented:
Alright.  I will make changes according to your suggestions.
Thank you for the feedback!

psalmuelle committed:
refactor: Increase size of projectDetails page

psalmuelle committed:
refactor: Rename enlarged image id

psalmuelle commented:
@tuxology. I have updated the UI according to your suggestions

- I have changed the cancel button's position to the image's top right corner.

- I have also styled the arrow to look bigger and decreased the background opacity.

- Finally, I increased the size of the image in-preview

Screenshot from a PC
![correction](https://user-images.githubusercontent.com/81099447/224347955-35c2bcca-c561-4d1b-a879-88f7e0fd62ae.jpg)

Tablet
![323](https://user-images.githubusercontent.com/81099447/224348030-876dd5eb-9c92-4f80-b077-ddf011460d7f.jpg)

Mobile
![22](https://user-images.githubusercontent.com/81099447/224348060-1d75a78a-9654-41b7-9463-35569f691237.jpg)

Is there any other thing that can be improved? 
Thank you!

tuxology commented:
Nice, looks nice for now, I'll review and send feedback @psalmuelle

psalmuelle commented:
Alright. Thank you!

psalmuelle committed:
Merge branch 'unstructuredstudio:master' into master

kamthamc commented on zubhub_frontend/zubhub/src/assets/js/styles/views/project_details/projectDetailsStyles.js:
```diff
@@ -1,3 +1,5 @@
+import zIndex from "@material-ui/core/styles/zIndex";
```
Use Prettier plugin to format files, the formatting seems to be off.

kamthamc replied:
And use styles instead of inline styles

kamthamc commented on zubhub_frontend/zubhub/src/assets/js/styles/views/project_details/projectDetailsStyles.js:
```diff
@@ -161,15 +163,32 @@ const styles = theme => ({
     height: 'auto',
   },
+  cancelEnlargedImageBtn:{
+   backgroundColor: '#F50057',
```
Use Theme for the colors

kamthamc commented on zubhub_frontend/zubhub/src/assets/js/styles/views/project_details/projectDetailsStyles.js:
```diff
@@ -161,15 +163,32 @@ const styles = theme => ({
+  cancelEnlargedImageBtn:{
+   backgroundColor: '#F50057',
+   position: 'absolute',
+   right: '6%',
```
I think the button should be have a constant size from the right?
cc @tuxology

kamthamc commented on zubhub_frontend/zubhub/src/assets/js/styles/views/project_details/projectDetailsStyles.js:
```diff
@@ -161,15 +163,32 @@ const styles = theme => ({
+   width: '34px',
+   height: '34px',
+   borderRadius: '100%',
+   border: '2px solid white',
```
Use themes for colors

kamthamc commented on zubhub_frontend/zubhub/src/assets/js/styles/views/project_details/projectDetailsStyles.js:
```diff
@@ -369,4 +388,31 @@ export function PrevArrow(props) {
+  return (
+    <div
+      className={className}
+      style={{ ...style, 
```
Better to use class for this and FYI the style/any prop spread should be at the end so we can override styles

kamthamc commented on zubhub_frontend/zubhub/src/views/project_details/ProjectDetails.jsx:
```diff
@@ -491,12 +507,57 @@ function ProjectDetails(props) {
+                {project.images.map(image => (
+                  <div>
+                    <img
+                      style={{
```
Use classes instead of inline styles

kamthamc replied:
And images need to have alt text, If the image doesn't have title or description properties add empty text for alt

kamthamc commented on zubhub_frontend/zubhub/src/views/project_details/ProjectDetails.jsx:
```diff
@@ -491,12 +507,57 @@ function ProjectDetails(props) {
+                open_enlarged_image_dialog: !open_enlarged_image_dialog,
+              })
+            }
+            aria-label="close"
```
aria labels should also be translated

psalmuelle replied:
I have made note of all corrections, Ill make changes right away. Thank you @kamthamc

psalmuelle committed:
Merge branch 'unstructuredstudio:master' into master

tuxology commented:
@psalmuelle Are we done with this and is this merge ready or are you working still on suggestions from @kamthamc ?
